### ISSUE TITLE
Power level and Moderation drop-in replacement - based on leagues

### ISSUE THREAD

**Issue by sk-alexis** [matrix-org/matrix-doc#2811]
DISCLAIMER
This text doesn't intend to be specific to a certain reputation algorithm or implementation just presenting the problem with the current perspective/implementation in a organized manner, and act has a working framework in which experimental solutions can be implemented on. In addition is important to refer this framework is compatible with the current structure and codebases to what relates to the 2 topics discussed underneath. the only part it replaces is the 0-100 power level feature, with a implementation discussed on the #matrix-reputation:matrix.org

So first of all this 2 problems are: power structure and moderation
On what relates to the organization and management of a room, and its members.

POWER MANAGEMENT:
PROBLEM: There are many issues in this realm, but at its core most relate to the fact that peers can have a power level in common, nor being distinct from each other for that matter.
SUBPROPUSAL: Each room member needs a different power level, and my proposal being to include a LIST/RANKING, this is by far the most intuitive and easy way to manage unique power levels for and by each room member, obviously being the person above you, in a higher power level than you.

MODERATION:
PROBLEM: I think it’s obvious given all the issue it has brought to the technology industry.
SUBPROPOUSAL: So far this area of moderation as focused in shutting down bad apples, instead of a system of merit, ICENTIVIZEvsPROIBITION dichotomy, that have proven to work much better (e.g. marketplaces). So the solution in moderation is to incentivize good behavior instead of to an extreme banning kicking people, with a merit/review system, you would give more popularity/visibility to popular members, especially with the growth of the network. This has been tried/done by the minds social network project, although I’m not a fan of minds, the core idea is great.
 
PROPOUSAL
joining these 2 problems

WHY are this 2 topics related
This relates to a deeper architectural problem if there isn’t any incentive to act in good faith but only fear of bad behavior and its repercussions, most people will not be interested in being productive. So moderation needs to be connected with power management, therefore most unsupported views need to be the ones acted upon, this probably would means the removal of kicks and bans in the beginning(maybe implement on P2P first for experimentation, this could be reimplemented on a referee system based on this implementation, but that is outside the reach of this proposal). Again moderation needs to be included, being that both are corelated, if only power management would be taken into consideration, people within a division could depromote (self-promotion is not possible) a good popular/supported member, introducing a problem with spam(notice that the opposite is also true, spam would mostly be solved by such system) so that this could be soved by aplying the promotions from others (this also could be expanded by +1 signs or just allowing positive emojis, and counting that as a value merit system this is outside the scope of this proposal again given is not a necessary thing for the system to work) basically self-balancing the this propoused framework based on merit.

SOLUTION
so you would keep the current role structure, and permission settings, but create DIVISIONS/LEAGUES within each role, basically a filter that instead of A-Z the member list, it would be divide in groups/roles/"powerlevels" each with a list/ranking of power based on reputation. In other words, basically what would change is only the system behind the current power level classification, instead of having 0-100 power level system you would have a similar structure to what of a sport division system, which the classification for each division would be according to merit, so every person would in fact have a different "level" on what comes to member management.

References
GVRN, DAO's, minds social network, most marketplace/rental/restaurant review systems

Acknowledgments
Please give all feedback you might have, and would really appreciate any refutations, to any part of this proposal, I invite criticism, to try to understand the holes in such logic/arguments

Notes
I believe this should be in conjunction or at least in synergy with the threading and identity/trust efforts.

And thank you for understanding, also the time and energy listening to this on the main chat or reading it here . |3^RELATIVISM. I usually boycott GitHub, but feel this is an important topic, so I made a concession, to check for updates check my libregit.org repository Cheers.

http://www.cawtech.demon.co.uk/outer-circle/core.html
http://lup.lub.lu.se/luur/download?func=downloadFile&recordOId=9020253&fileOId=9020270

**Comment by sk-alexis** [comment]
this probably will be transformed in a spec proposal didn't realized that gatehub doesn't have feature request like gitlab/libregit, once I have some available time, or if any technical writer is interested in it. As @turt2live suggested according to https://matrix.org/docs/spec/proposals. Cheers, any fedback would be usefull in the mean while.

**Comment by sk-alexis** [comment]
@turt2live could you include moderation on the title as well, given is also the basis for that.
cheers

**Comment by richvdh** [comment]
I think the TL;DR here is that users should be maintained in a sorted list denoting their relative power levels.

Feel free to make a spec change proposal, but I don't think this is a direction we are likely to take the protocol. I'm inclined to close this unless other members of the spec team want to weigh in.

**Comment by sk-alexis** [comment]
http://www.cawtech.demon.co.uk/outer-circle/core.html related to this topic, implementation

**Comment by sk-alexis** [comment]
@richvdh Yes but not exactly, the basic idea is to do like 'sport divisions' according to power levels available, so 3 at the moment, on which each division is organized by a list.

**Comment by sk-alexis** [comment]
Also related http://lup.lub.lu.se/luur/download?func=downloadFile&recordOId=9020253&fileOId=9020270

**Comment by sk-alexis** [comment]
@richvdh also I think it would be a shame to close this down, this feauture request, given I heard many times a lot of interest in similar system from some of the core members, including but not exclusively @turt2live and @ara4n, and also a lot of chat about similar system on the #reputation:matrix.org room. If you feel this doesn't accurately describe such endeavor/topics please feel free to make suggestions/refutations/improvements to such text. Cheers.

**Comment by sk-alexis** [comment]
@richvdh I will close this once spec propousal is complited though, you or @turt2live could add the framework tag as well. cheers

**Comment by turt2live** [comment]
Please stop pinging me.

The idea is interesting, though I agree that the protocol is unlikely to take this direction.

**Comment by sk-alexis** [comment]
owh that is unnecessary, wasn't pinging you for either acceptance or attention, I couldn't care less if this is implemented in matrix or not, so pleaso dont jump in to conclusion, or let your ego get in here like I pointed out yesterday. Im more interested in hearing people feedback so improvements can be added to upstream GVRN project. If I pinged you @turt2live is because I can not add framework tag. Holy fu... you most not be having a good day.
